proxy/config.go

```diff
@@ -0,0 +1,79 @@
+package config
+
+import (
+	"io/ioutil"
+
+	"gopkg.in/yaml.v2"
+)
+
+type AMQPConfig struct {
+	URL      string `yaml:"url,omitempty"`
+	Exchange string `yaml:"exchange,omitempty"`
+}
+
+// TLSConfig configures the options for TLS connections.
+type TLSConfig struct {
+	// The CA cert to use for the targets.
+	CAFile string `yaml:"ca_file,omitempty"`
+	// The client cert file for the targets.
+	CertFile string `yaml:"cert_file,omitempty"`
+	// The client key file for the targets.
+	KeyFile string `yaml:"key_file,omitempty"`
+	// Used to verify the hostname for the targets.
+	ServerName string `yaml:"server_name,omitempty"`
+	// Disable target certificate validation.
+	InsecureSkipVerify bool `yaml:"insecure_skip_verify"`
+
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline"`
+}
+
+type TargetGroup struct {
+	Targets []string
+}
+
+type ScrapeConfig struct {
+	// The job name to which the job label is set by default.
+	JobName    string     `yaml:"job_name"`
+	AMQPConfig AMQPConfig `yaml:"amqp_config,omitempty"`
+	// TLSConfig to use to connect to the targets.
+	TLSConfig TLSConfig `yaml:"tls_config,omitempty"`
```

Comment:
Nope.
Go aligns stuff itself (with gofmt). Comments and blank lines break continuation.